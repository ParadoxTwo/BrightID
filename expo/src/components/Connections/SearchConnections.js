// @flow

import * as React from 'react';
import { TextInput, TouchableOpacity, StyleSheet, View } from 'react-native';
import { connect } from 'react-redux';
import Ionicon from 'react-native-vector-icons/Ionicons';
import Octicons from 'react-native-vector-icons/Octicons';
import { setSearchParam } from '../../actions';

/**
 * Search Bar in the Connections Screen
 * TODO: Add functionality for the Ionicons
 * TODO: add search filter in redux actions
 */

type LocalProps = {
  sortable: boolean,
};

class SearchConnections extends React.Component<Props & LocalProps> {
  componentWillUnmount() {
    // reset search Param
    const { dispatch } = this.props;
    dispatch(setSearchParam(''));
  }

  render() {
    const { sortable } = this.props;
    return (
      <View style={styles.container}>
        <TouchableOpacity style={styles.searchIcon}>
          <Octicons size={26} name="search" color="#333" />
        </TouchableOpacity>
        <TextInput
          onChangeText={(value) => this.props.dispatch(setSearchParam(value))}
          style={styles.searchField}
          placeholder="Search Connections"
          autoCapitalize="words"
          autoCorrect={false}
          textContentType="name"
          underlineColorAndroid="transparent"
        />
        {sortable && (
          <TouchableOpacity
            onPress={() => {
              const { navigation } = this.props;
              navigation.navigate('SortingConnections');
            }}
            style={styles.optionsIcon}
          >
            <Ionicon size={30} name="ios-options" color="#333" />
          </TouchableOpacity>
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginTop: 10,
    width: '90%',
    borderColor: '#ccc',
    borderWidth: 1,
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: '#fff',
  },
  searchIcon: {
    marginLeft: 10,
    marginRight: 12,
    marginTop: 5,
  },
  optionsIcon: {
    marginLeft: 10,
    marginRight: 8.8,
    marginTop: 5,
  },
  searchField: {
    fontFamily: 'ApexNew-Book',
    fontSize: 16,
    marginTop: 3.1,
    marginLeft: 23,
    flex: 1,
    fontWeight: 'normal',
    fontStyle: 'normal',
    letterSpacing: 0,
  },
});

export default connect()(SearchConnections);
