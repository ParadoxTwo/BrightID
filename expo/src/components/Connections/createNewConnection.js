// @flow

import { Alert } from 'react-native';
import { store } from '../../store';
import { addConnection } from '../../actions/fakeContact';

export const createNewConnection = (navigation: navigation) => () => {
  if (__DEV__) {
    Alert.alert(
      'New Connection',
      'Would you like simulate adding a new connection?',
      [
        {
          text: 'Cancel',
          onPress: () => console.log('Cancel Pressed'),
          style: 'cancel',
        },
        {
          text: 'Sure',
          onPress: () => {
            store.dispatch(addConnection(navigation));
          },
        },
      ],
      { cancelable: true },
    );
  }
};
