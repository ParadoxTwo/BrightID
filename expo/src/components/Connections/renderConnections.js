import * as React from 'react';
import { FlatList, StyleSheet, Text, View } from 'react-native';
// import Spinner from 'react-native-spinkit';

export const renderListOrSpinner = (comp) => {
  const { connections } = comp.props;
  if (comp.state && comp.state.loading) {
    return <View />;
  } else if (connections.length > 0) {
    return (
      <FlatList
        style={styles.connectionsContainer}
        data={comp.filterConnections()}
        keyExtractor={({ id }, index) => id + index}
        renderItem={comp.renderConnection}
      />
    );
  } else {
    return (
      <View>
        <Text style={styles.emptyText}>No connections</Text>
      </View>
    );
  }
};

const styles = StyleSheet.create({
  connectionsContainer: {
    flex: 1,
  },
  emptyText: {
    fontFamily: 'ApexNew-Book',
    fontSize: 20,
  },
});
