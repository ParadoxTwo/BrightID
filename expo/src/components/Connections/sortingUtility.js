// @flow
import { setConnections, setConnectionsSort } from '../../actions';

export const types = {
  byNameAscending: 'BY_NAME_ASCENDING',
  byNameDescending: 'BY_NAME_DESCENDING',
  byScoreAscending: 'BY_SCORE_ASCENDING',
  byScoreDescending: 'BY_SCORE_DECENDING',
  byDateAddedAscending: 'BY_DATE_ADDED_ASCENDING',
  byDateAddedDescending: 'BY_DATE_ADDED_DESCENDING',
};

export const sortByNameAscending = () => (
  dispatch: dispatch,
  getState: getState,
) => {
  const { connections } = getState();
  let list = connections.slice();
  list.sort((a, b) => b.name.localeCompare(a.name));
  dispatch(setConnections(list));
  dispatch(setConnectionsSort(types.byNameAscending));
};

export const sortByNameDescending = () => (
  dispatch: dispatch,
  getState: getState,
) => {
  const { connections } = getState();
  let list = connections.slice();
  list.sort((a, b) => a.name.localeCompare(b.name));
  dispatch(setConnections(list));
  dispatch(setConnectionsSort(types.byNameDescending));
};

export const sortByScoreAscending = () => (
  dispatch: dispatch,
  getState: getState,
) => {
  const { connections } = getState();
  let list = connections.slice();
  list.sort((a, b) => a.score - b.score);
  dispatch(setConnections(list));
  dispatch(setConnectionsSort(types.byScoreAscending));
};

export const sortByScoreDescending = () => (
  dispatch: dispatch,
  getState: getState,
) => {
  const { connections } = getState();
  let list = connections.slice();
  list.sort((a, b) => b.score - a.score);
  dispatch(setConnections(list));
  dispatch(setConnectionsSort(types.byScoreDescending));
};

export const sortByDateAddedAscending = () => (
  dispatch: dispatch,
  getState: getState,
) => {
  const { connections } = getState();
  let list = connections.slice();
  list.sort((a, b) => a.connectionDate - b.connectionDate);
  dispatch(setConnections(list));
  dispatch(setConnectionsSort(types.byDateAddedAscending));
};

export const sortByDateAddedDescending = () => (
  dispatch: dispatch,
  getState: getState,
) => {
  const { connections } = getState();
  let list = connections.slice();
  list.sort((a, b) => b.connectionDate - a.connectionDate);
  dispatch(setConnections(list));
  dispatch(setConnectionsSort(types.byDateAddedDescending));
};

export const defaultSort = () => (dispatch: dispatch, getState: getState) => {
  const { connectionsSort } = getState();
  switch (connectionsSort) {
    case types.byNameAscending:
      dispatch(sortByNameAscending());
      break;
    case types.byNameDescending:
      dispatch(sortByNameDescending());
      break;
    case types.byDateAddedAscending:
      dispatch(sortByDateAddedAscending());
      break;
    case types.byDateAddedDescending:
      dispatch(sortByDateAddedDescending());
      break;
    case types.byScoreAscending:
      dispatch(sortByScoreAscending());
      break;
    case types.byScoreDescending:
      dispatch(sortByScoreDescending());
      break;
    default:
      dispatch(sortByDateAddedDescending());
  }
};
