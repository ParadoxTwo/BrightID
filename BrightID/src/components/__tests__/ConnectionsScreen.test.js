import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { ConnectionsScreen } from '../Connections/ConnectionsScreen';

// you can mock any middlewares here if necessary

const props = {
  dispatch: jest.fn(),
  navigation: { navigate: jest.fn() },
  searchParam: '',
  connections: [],
};

const renderer = new ShallowRenderer();

describe('Testing ConnectionsScreen', () => {
  it('renders blank connection screen', () => {
    renderer.render(<ConnectionsScreen {...props} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
