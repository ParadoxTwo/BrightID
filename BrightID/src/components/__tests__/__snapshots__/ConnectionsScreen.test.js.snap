// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ConnectionsScreen renders blank connection screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fdfdfd",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fdfdfd",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginTop": 8,
        }
      }
    >
      <SearchConnections
        navigation={
          Object {
            "navigate": [MockFunction],
          }
        }
        sortable={true}
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fdfdfd",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "marginTop": 8,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "fontFamily": "ApexNew-Book",
                "fontSize": 20,
              }
            }
          >
            No connections
          </Text>
        </View>
      </View>
    </View>
    <FloatActionButton
      onPress={[Function]}
    />
  </View>
  <Connect(BottomNav)
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
    style={
      Object {
        "flex": 0,
      }
    }
  />
</View>
`;
